components/ColumnSelect.tsx

```diff
@@ -0,0 +1,85 @@
+import { action, computed, makeObservable } from 'mobx';
+import { observer } from 'mobx-react';
+import * as React from 'react';
+import Select from 'react-select';
+
+type Option = {
+    value: string;
+    label: string | JSX.Element;
+    disabled?: boolean;
+};
+
+export type ColumnVisibilityDef = {
```

Comment:
nitpicky but I think its more of a specification than a definition, so I would say its `ColumnVisibilitySpec`. or even just `ColumnVisibility` would be fine.